### PR TITLE
FHB-842:  login bug

### PR DESCRIPTION
**Pull request by StuwiiDev**
Difficult to diagnose issue as the sessions are deleted as soon as someone else successfully logs in.
The theory is that the service was able to post a new session with the same SID as one that already existed. As the error only seems to occur in the morning, it's possible that it's old expired sessions.
This update is to ensure expired ones are removed first before attempting a new one.

Monitoring app insights after this goes in can tell us if it's working or not.

### CODE REVIEW

File: src/service/idam-api/src/FamilyHubs.Idam.Core/Commands/Delete/DeleteExpiredUserSessionsCommand.cs [hunk 1]
```diff
@@ -31,19 +28,19 @@ public async Task<bool> Handle(DeleteExpiredUserSessionsCommand request, Cancell
...
-            DateTime sessionExpiryTime = DateTime.UtcNow.AddSeconds((int)-cleanupInterval);
-            List<UserSession> entities =
+            var sessionExpiryTime = DateTime.UtcNow.AddSeconds((int)-cleanupInterval);
+            var entities =
                 await _dbContext.UserSessions.Where(r => r.LastActive < sessionExpiryTime).ToListAsync(cancellationToken);
 
             _logger.LogInformation("Found: {CountExpired} expired sessions.", entities.Count);
 
-            if (entities.Any())
+            if (entities.Count != 0)
```
Comment (stevesatdfe) [inline src/service/idam-api/src/FamilyHubs.Idam.Core/Commands/Delete/DeleteExpiredUserSessionsCommand.cs:43]: Probably > as Count will never be less than zero

**Review by stevesatdfe — CHANGES_REQUESTED** [review]

File: src/service/idam-api/src/FamilyHubs.Idam.Core/Commands/Delete/DeleteExpiredUserSessionsCommand.cs (hunk 1 above)
Comment (StuwiiDev) [reply src/service/idam-api/src/FamilyHubs.Idam.Core/Commands/Delete/DeleteExpiredUserSessionsCommand.cs:43]: The != 0 is the direct suggested replacement of Any() but yes that also makes sense

**Review by stevesatdfe — DISMISSED** [review]

File: src/service/idam-api/tests/FamilyHubs.Idam.Api.UnitTests/FamilyHubs.Idam.Api.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+
+        <IsPackable>false</IsPackable>
+        <IsTestProject>true</IsTestProject>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="coverlet.collector" Version="6.0.0"/>
```
Comment (Zac-Digital) [inline src/service/idam-api/tests/FamilyHubs.Idam.Api.UnitTests/FamilyHubs.Idam.Api.UnitTests.csproj:13]: Should be centralised in `Directory.Packages.props`

Comment (StuwiiDev) [reply src/service/idam-api/tests/FamilyHubs.Idam.Api.UnitTests/FamilyHubs.Idam.Api.UnitTests.csproj]: updated

**Review by Zac-Digital — APPROVED** [review]

**Review by stevesatdfe — APPROVED** [review]

**Pull request merged**